pkg/podclient/baseclient.go

```diff
@@ -0,0 +1,69 @@
+package podclient
```

Comment:
I noticed there was some overlap in what this needed to do with k8s and what the worker does, so I pulled out some parts into a separate package. I kept it very minimal as I'm not entirely sure if it will stay that way.